sdk/helper/awsutil/generate_credentials.go

```diff
@@ -54,6 +59,20 @@ func (c *CredentialsConfig) GenerateCredentialChain() (*credentials.Credentials,
 			"static AWS client credentials haven't been properly configured (the access key or secret key were provided but not both)")
 	}
 
+	roleARN := os.Getenv("AWS_ROLE_ARN")
+	tokenPath := os.Getenv("AWS_WEB_IDENTITY_TOKEN_FILE")
+	if roleARN != "" && tokenPath != "" {
+		sessionName := os.Getenv("AWS_ROLE_SESSION_NAME")
```

Comment:
According to [this doc](https://docs.aws.amazon.com/STS/latest/APIReference/API_AssumeRole.html), this is a required parameter. Since all three are required, could we move this up and add it to the `if` check before heading into this code?

Reply:
You are right, I moved the retrieval of the env var up and added it to the `if`check.